Lib/_pydatetime.py

```diff
@@ -1241,6 +1241,11 @@ def weekday(self):
         "Return day of the week, where Monday == 0 ... Sunday == 6."
         return (self.toordinal() + 6) % 7
 
+    def timestamp(self):
+        """Return POSIX timestamp as float (midnight UTC)."""
+        epoch = date(1970, 1, 1)
```

Comment:
Maybe we could cache the epoch.toordinal() value? or just hardcode it (as it's UTC)